### PR TITLE
update the prefix code

### PR DESCRIPTION
**Pull request by pikaninja**
This pr updates get_prefix to allow extra spaces(ie `p! ping`) and also case insensitive for custom prefixes. example show below with the prefix "poketwo"
image

### CODE REVIEW

**Review by NCPlayz — APPROVED** [review]

File: cogs/bot.py [hunk 1]
```diff
@@ -188,7 +188,16 @@ async def on_guild_join(self, guild):
         )
         await channel.send(embed=embed)
 
-    async def determine_prefix(self, guild):
+    def case_insensitive_prefix(self, prefix, content):
+        match = re.match(f"^({prefix}).*", content, flags=re.I)
```
Comment (oliver-ni) [inline cogs/bot.py:192]: prefix needs to be regex escaped

File: cogs/bot.py [hunk 2]
```diff
@@ -199,18 +208,16 @@ async def determine_prefix(self, guild):
...
-            "P!",
-            self.bot.user.mention + " ",
-            self.bot.user.mention[:2] + "!" + self.bot.user.mention[2:] + " ",
-        ]
+                prefix = self.bot.prefixes[guild.id]
+
+        if message:
+            prefix = self.case_insensitive_prefix(prefix, message.content)
+            if not prefix:
+                return commands.when_mentioned(self.bot, message)
```
Comment (oliver-ni) [inline cogs/bot.py:216]: can you return a list in these two places? parts of my code rely on that. (lol)

**Review by oliver-ni — CHANGES_REQUESTED** [review]

File: cogs/bot.py (hunk 1 above)
Comment (NCPlayz) [reply cogs/bot.py:192]: cannot believe i glossed over this. definitely an issue haha!

**Review by NCPlayz — APPROVED** [review]
Seems good now.

**Comment by pikaninja** [comment]
outdated

**Pull request closed without merging**